### ISSUE TITLE
Previous [cached?] image names don't change to new image

### ISSUE THREAD

**Issue by dylanh724** [discordapp/discord-rpc#155]
This is reproduced with small images; it's safe to assume this also happens with large images, but untested.

**To reproduce:**

1 - Upload "somesmallimage"
2 - Run once with this image to make it show ( Unsure if this is necessary )
3 - Delete the image
4 - Upload a new image **named the same**
5 - Future tests will still show **the old [cached?] image**

**Comment by msciotti** [comment]
This is true—we do cache things pretty hard. If you open your Discord console when you send a `SET_ACTIVITY` payload, you'll see that we're swapping out the string you send for an image id. In this case, we're grabbing the same image id as the "old" one, because or backend says "Hey, same key, same id, easy."

I don't really have a better solution for you right now. I could see a world in which we do something like...idk, adding a `debug` field in the `SET_ACTIVITY` payload that ignores cache? That seems like an abuse vector though.

Maybe @night can weigh in on a good way to develop around this for now. Even trying it on a localdev client with caching disabled, I get the same results as you.

**Comment by dylanh724** [comment]
Is it possible to wipe cache as new assets are uploaded? By the way, the uploader allows for multiple uploads of the same name, which I don't know if should be posted here or not. Intuitively it would replace the old one and temporarily wipe cache for that string. If that's possible, anyway.

Or - what you can do with a cache is only allow cache clearing once every x amount of time because it's only going to be used during development. It's the same thing Cloudflare does! You can only clear cache once per 15 minutes or something.

**Comment by msciotti** [comment]
Sure, there's lots we could do in the long run. Right now, we don't have the infrastructure to support the difference between an app "in development" vs. "released" unfortunately. That, and allowing users to explicitly bypass the cache could be an abuse vector against our own network/CDN. 

Operating on the scale that we do, we always prefer to cache hard when we can. A game like League of Legends, for example, could be accessing the same asset millions of times simultaneously, and 99.99% of the time, that asset won't change.

I will let night speak as to the feasibility of either solving this totally valid concern on our end, or a workaround that you could implement on your side.

**Comment by Almighty-Alpaca** [comment]
@dylanh724 I have to admit it's not a great solution but for testing purposes you can delete `%appdata%/discord/cache` and then restart discord.

**Comment by MFernflower** [comment]
I suppose this is related but when an image is deleted is there still a resized preview copy stored in the cache of the machines who were in the channel it was sent out to?
